.github/workflows/trunk-build.yml

```diff
@@ -0,0 +1,51 @@
+name: Build on `trunk`
+
+on:
+  push:
```

Comment:
Do we want this to also run on just _any_ push to _any_ branch? Or only on pushes to `main` and pulls?

Reply:
I'd say only on push events to `main`, I'll update the action!

Reply:
For PRs I'd keep any branch though, does that sound good?